From 6814f2b965f77b07c4a2ac6fc8faa573c245448c Mon Sep 17 00:00:00 2001
From: Rick Farrington <rfarrington@marvell.com>
Date: Mon, 16 Nov 2020 22:29:15 -0500
Subject: [PATCH 678/767] ghes_edac: remove erroneous error message regarding
 DIMM info

commit 616827e4f0530cb5654c900fb50202c666d566b7 from
git@git.assembla.com:cavium/WindRiver.linux.git

When used w/OCTEONTX2_SDEI_GHES, the ghes_edac DIMM information in
any error reports is correct (i.e. SMBIOS/DMI info is unnecessary).
Therefore, do not display error message regarding incorrect DIMMs
when OCTEONTX2_SDEI_GHES is present.

Change-Id: I8d9aec30d18b7ad774d16be112b9d32072e5ed71
Signed-off-by: Rick Farrington <rfarrington@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/40191
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
---
 drivers/edac/ghes_edac.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/edac/ghes_edac.c b/drivers/edac/ghes_edac.c
index 60d1d93bd195..f1da41cbd11d 100644
--- a/drivers/edac/ghes_edac.c
+++ b/drivers/edac/ghes_edac.c
@@ -533,9 +533,12 @@ int ghes_edac_register(struct ghes *ghes, struct device *dev)
 	mci->dev_name = "ghes";
 
 	if (fake) {
+/* OCTEONTX2_SDEI_GHES supports correct DIMM reporting w/o DMI/SMBIOS info */
+#ifndef CONFIG_OCTEONTX2_SDEI_GHES
 		pr_info("This system has a very crappy BIOS: It doesn't even list the DIMMS.\n");
 		pr_info("Its SMBIOS info is wrong. It is doubtful that the error report would\n");
 		pr_info("work on such system. Use this driver with caution\n");
+#endif
 	} else if (idx < 0) {
 		pr_info("This EDAC driver relies on BIOS to enumerate memory and get error reports.\n");
 		pr_info("Unfortunately, not all BIOSes reflect the memory layout correctly.\n");
-- 
2.31.1

